Source/Particles/Collision/BinaryCollision/PhotonPhoton/TwoProductUtil.H

```diff
@@ -0,0 +1,142 @@
+
+        // Energy squared of each of the two colliding photons in the center of momentum frame,
+        // calculated using the Lorentz invariance of the total four-momentum norm
+        const amrex::ParticleReal E_star_sq = one_half_pr*c_sq*p1_in*p2_in*(one_pr - cos_ang);
+
+        // Square of the norm of the momentum of the products in the center of mass frame
+        // Formula obtained by inverting E^2 = p^2*c^2 + m^2*c^4 in the COM frame for each particle:
+        // p_star_sq = E_star_sq/c_sq - me*me*c_sq;
+        // The expression below is specifically written in a form that avoids returning
+        // small negative numbers due to machine precision errors, for low-energy particles
+        const amrex::ParticleReal E_ratio = std::sqrt(E_star_sq)/(2._prt*me*c_sq);
+        const amrex::ParticleReal p_star_sq = me*me*c_sq * ( pow2(2._prt*E_ratio) - 1._prt );
+
+        // Compute momentum of first product in the center of mass frame, assuming isotropic
+        // distribution
+        amrex::ParticleReal px_star, py_star, pz_star;
+        ParticleUtils::RandomizeVelocity(px_star, py_star, pz_star, std::sqrt(p_star_sq),
+                                         engine);
+
+        // Next step is to convert momenta to lab frame
+        amrex::ParticleReal p1x_out, p1y_out, p1z_out;
+        // Preliminary calculation: compute velocity of the center of momentum frame:
+        // v = (p1 + p2) / | p1 + p2 | * c
+        const amrex::ParticleReal vcx    = (p1x_in+p2x_in) * PhysConst::c / (p1_in + p2_in);
+        const amrex::ParticleReal vcy    = (p1y_in+p2y_in) * PhysConst::c / (p1_in + p2_in);
+        const amrex::ParticleReal vcz    = (p1z_in+p2z_in) * PhysConst::c / (p1_in + p2_in);
+        const amrex::ParticleReal vc_sq  = vcx*vcx + vcy*vcy + vcz*vcz;
+
+        // Convert momentum of first product to lab frame, using equation (13) of F. Perez et al.,
+        // Phys.Plasmas.19.083104 (2012) (which is a regular Lorentz transformation)
+        if ( vc_sq > std::numeric_limits<amrex::ParticleReal>::min() )
+        {
+            const amrex::ParticleReal gc = 1._prt / std::sqrt( 1._prt - vc_sq*inv_csq );
+            const amrex::ParticleReal g_star = std::sqrt(1._prt + p_star_sq / (me*me*c_sq));
+            const amrex::ParticleReal vcDps = vcx*px_star + vcy*py_star + vcz*pz_star;
+            const amrex::ParticleReal factor0 = (gc-1._prt)/vc_sq;
+            const amrex::ParticleReal factor = factor0*vcDps + me*g_star*gc;
+            p1x_out = px_star + vcx * factor;
+            p1y_out = py_star + vcy * factor;
+            p1z_out = pz_star + vcz * factor;
+        }
+        else // If center of mass velocity is zero, we are already in the lab frame
+        {
+            p1x_out = px_star;
+            p1y_out = py_star;
+            p1z_out = pz_star;
+        }
+
+        // Compute momentum of beryllium in lab frame, using total momentum conservation
```

Comment:
Not sure which species this is supposed to be:
```suggestion
        // Compute momentum of electron/positron in lab frame, using total momentum conservation
```